code/game/turfs/simulated/floor.dm

```diff
@@ -166,7 +166,9 @@
 		if(user && !silent)
 			to_chat(user, "<span class='danger'>You remove the floor tile.</span>")
 		if(floor_tile && make_tile)
-			new floor_tile(src)
+			var/obj/item/I = new floor_tile(src)
```

Comment:
these parts are the only good parts.
if you make a new pr with just the "crowbarring up ice remove it from the turf and makes ice floor tiles" parts (+ making it so if you lay that floor tile the new turf inherits the ice) then that'd probably be accepted for ~rule of cool~